### ISSUE TITLE
Introduce explicit caption positioning

### ISSUE THREAD

**Issue by melchoyce** [WordPress/gutenberg#12997]
**Is your feature request related to a problem? Please describe.**
Captions on images are shown centered in Gutenberg, but theme styles can override this. There's no way to explicitly set an alignment onto the caption, without writing additional CSS to override your theme. 

**Describe the solution you'd like**
Introduce a text alignment setting to captions, and have any explicit change in alignment you make override theme styles.

**Describe alternatives you've considered**
More specific center styles on the Gutenberg block itself.

**See also**
- https://wordpress.org/support/topic/image-caption-not-centering/
- https://wordpress.org/support/topic/image-caption-poorly-placed-on-page-gutenberg-annoyance/

**Comment by paaljoachim** [comment]
Hey Mel

I added on to this (closed) issue: https://github.com/WordPress/gutenberg/issues/534
As I believe that additional options could be added to the styling caption dropdown.

**Comment by joyously** [comment]
> Captions on images are shown centered in Gutenberg,

That style should be in the styles that are optional, not default.

**Comment by richtabor** [comment]
@melchoyce What do you think of adding the AlignmentToolbar to the InlineToolbar, like [this](https://github.com/WordPress/gutenberg/pull/15554#issuecomment-506818526) exploration in #15554?

![captions](https://user-images.githubusercontent.com/1813435/60743811-21c8a900-9f41-11e9-86b5-691938bf0aa0.jpg)

Or this, with the AlignmentToolbar on the left, as it is in the toolbar of the Paragraph block? 
![alignment-left](https://user-images.githubusercontent.com/1813435/60743937-aadfe000-9f41-11e9-9bc4-41d627dff9c8.jpg)

**Comment by paaljoachim** [comment]
It would be great to add alignment directly to the captions area, as I have noticed that the caption can be off center on various images that I add. Adding the alignment inside the captions area would let the user decide where the caption should be seen.

**Comment by melchoyce** [comment]
Yeah, exactly that @richtabor.

**Comment by melchoyce** [comment]
Fun caption position from [Snow Fall](http://www.nytimes.com/projects/2012/snow-fall/index.html#/?part=descent-begins): 
![image](https://user-images.githubusercontent.com/2846578/60910654-6315cd00-a24f-11e9-815a-c8630742bd16.png)

**Comment by LifeHack3r** [comment]
Facing similar issues!

**Comment by plittlefield** [comment]
Yeah, still not fixed. I've got around it with some CSS...

```
.wp-block-image figcaption {
	text-align: center;
}
```

Paully

**Comment by mark2090** [comment]
> 
> 
> Yeah, still not fixed. I've got around it with some CSS...
> 
> ```
> .wp-block-image figcaption {
> 	text-align: center;
> }
> ```
> 
> Paully

> 
> Yeah, still not fixed. I've got around it with some CSS...
> 
> ```
> .wp-block-image figcaption {
> 	text-align: center;
> }
> ```
> 

The CSS you gave will cause all figcaptions of images to center. If the image is always centered on the website, then that is fine, but if you want to be specific to the page and not make it global, use the "Advanced" settings" in the right column of the block editor for that image block and add a specific CSS class name (at the bottom of the column). This way, you can use that class name only when you want the caption to be centered, leaving the rest at the per the default of left align. 

For the Custom CSS, you do not need to add the tag name (figcaption) as part of the selector. It works fine with just the class name you assign it.

Example:
```
_Right Column Image Block settings_
"Additional CSS class(es)" >   captionAlign

_Custom CSS (or Additional CSS)_
.captionAlign { text-align: center; }
```

**Still not Fixed??**
With no disrespect, I say the following with "best interests of the community" in mind including all users of WordPress.

This figcaption text centering issue, is the kind of thing that is fundamental, basic and not something that is extra in any sense of the term. 
I got to this bug report from a WordPress.org Support post that is 18 months old, which makes that about January 2019. There have been several WordPress updates since and this bug is still not fixed. I am currently working a website using the block editor and had to use Custom CSS to temporarily fix this issue. 

I am sure, if money depended on this, the bug would have been fixed within a month after it was first reported, that is what plugin and theme developers do.

**Comment by mtias** [comment]
This could use an updated design with the latest set of tools for clarity :) cc @kellychoffman

**Comment by paaljoachim** [comment]
I am wondering if we should initially at Image Caption alignment into the Global Styles panel as suggested in this issue:
https://github.com/WordPress/gutenberg/issues/27480

**Comment by kellychoffman** [comment]
In the meantime, an updated design that includes alignment option: 
<img width="633" alt="Screen Shot 2021-03-16 at 7 11 26 PM" src="https://user-images.githubusercontent.com/5835847/111404197-95660780-868b-11eb-8349-4c04e013d08a.png">

**Comment by plittlefield** [comment]
That, is superb.

**Comment by melchoyce** [comment]
I wonder if we should be using the text alignment icons instead of the block alignment ones? Since the block alignments would have wide/full instead of left/right.

**Comment by mtias** [comment]
Yes, I think it should be text alignment. I'm unsure if it should go on the right or the left, but if on the right it should be after the chevron (which indicates extra rich text formatting tools so it's tied to bold, italics, link).

**Comment by kellychoffman** [comment]
Ha! I feel dumb. Yes, 100% we should use text alignment.

<img width="673" alt="Screen Shot 2021-03-17 at 5 03 49 PM" src="https://user-images.githubusercontent.com/5835847/111554079-dff68b00-8742-11eb-8ae5-d249b53302de.png">

**Comment by paaljoachim** [comment]
Just thought I would mention it....
A change we need to make is having the caption align to the image, and not the boundary box surrounding the image.

**Comment by paaljoachim** [comment]
@Mamaduka George has made a PR here:
https://github.com/WordPress/gutenberg/pull/31814

Yahhh!

**Comment by kraftner** [comment]
#31814 from the previous comment has been closed in favor of #31823

**Comment by westonruter** [comment]
Just coming across this old issue here. I closed #19975 as a duplicate of this one.

In a recent blog post of mine, I wanted to have the captions center-aligned. I was able to achieve this with "Align center" in the Image block. However, as soon as I wanted to use the "Wide width" or "Full width", then I had no choice but to have the text left-justified, even when this results in a very poor rendering when "Full width" is used, since the text butts up against the left side of the screen:

Alignment | Screenshot
--|--
Align none | <img width="1015" height="481" alt="Image" src="https://github.com/user-attachments/assets/a951cb24-fd0c-4bf8-a737-edfa1d2a0a84" />
Align center | <img width="1014" height="486" alt="Image" src="https://github.com/user-attachments/assets/241be857-6973-44ca-b364-eec720a48594" />
Wide width | <img width="1015" height="656" alt="Image" src="https://github.com/user-attachments/assets/4afb9519-31cf-4a96-8211-05605370fd44" />
Full width | <img width="1016" height="716" alt="Image" src="https://github.com/user-attachments/assets/6d2bc50f-4d60-4a07-8cb5-41803d18bae4" />

This caption issue is not the Image block, of course. It is actually worse for the Table block, because for it the "Center align" doesn't cause the caption to be center-aligned.

Alignment | Screenshot
--|--
Align none | <img width="1015" height="148" alt="Image" src="https://github.com/user-attachments/assets/4a4460d6-06f4-4a5a-bf79-5f741fa70a4e" />
Align center | <img width="1015" height="150" alt="Image" src="https://github.com/user-attachments/assets/0f69f683-c57a-42c8-a579-3a461f1e7322" />
Wide width | <img width="1016" height="150" alt="Image" src="https://github.com/user-attachments/assets/978bb28e-5324-4e61-953e-de340cfd9e37" />
Full width | <img width="1017" height="144" alt="Image" src="https://github.com/user-attachments/assets/e2d58345-4102-4799-813c-7224084728d3" />

It's not clear what the difference is between “Align none” and “Align center” for the Table block, while at least for the Image block it causes the caption to be aligned differently.

**Comment by sabernhardt** [comment]
Could the Image block have a text alignment control, in addition to the block alignment?
